light-node/src/rpc.rs

```diff
@@ -0,0 +1,240 @@
+/// expose a handle in the future.
+pub fn run<H>(server: Server<H>, addr: &str) -> Result<(), error::Error>
+where
+    H: Handle + Send + Sync + 'static,
+{
+    let mut io = IoHandler::new();
+    io.extend_with(server.to_delegate());
+
+    let srv = ServerBuilder::new(io)
+        .cors(DomainsValidation::AllowOnly(vec![
+            AccessControlAllowOrigin::Any,
+        ]))
+        .start_http(&addr.parse().map_err(error::Kind::from)?)
+        .map_err(error::Kind::from)?;
+
+    srv.wait();
+
+    Ok(())
+}
+
+mod sealed {
+    use jsonrpc_core::futures::future::{self, Future, FutureResult};
+    use jsonrpc_core::types::Error;
+    use jsonrpc_derive::rpc;
+
+    use tendermint_light_client::supervisor::Handle;
+    use tendermint_light_client::types::LightBlock;
+
+    #[rpc]
+    pub trait Rpc {
+        /// Returns the latest trusted block.
+        #[rpc(name = "state")]
+        fn state(&self) -> FutureResult<Option<LightBlock>, Error>;
+    }
+
+    pub use self::rpc_impl_Rpc::gen_client::Client;
+
+    pub struct Server<H>
+    where
+        H: Handle + Send + Sync,
+    {
+        handle: H,
+    }
+
+    impl<H> Server<H>
+    where
+        H: Handle + Send + Sync,
+    {
+        pub fn new(handle: H) -> Self {
+            Self { handle }
+        }
+    }
+
+    impl<H> Rpc for Server<H>
+    where
+        H: Handle + Send + Sync + 'static,
+    {
+        fn state(&self) -> FutureResult<Option<LightBlock>, Error> {
+            let res = self
+                .handle
+                .latest_trusted()
+                .map_err(|_e| Error::internal_error());
+
+            future::result(res)
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use futures::compat::Future01CompatExt as _;
+    use jsonrpc_core::futures::future::Future;
+    use jsonrpc_core::IoHandler;
+    use jsonrpc_core_client::transports::local;
+    use pretty_assertions::assert_eq;
+
+    use tendermint_light_client::errors::Error;
+    use tendermint_light_client::supervisor::Handle;
+    use tendermint_light_client::types::LightBlock;
+
+    use super::{Client, Rpc as _, Server};
+
+    #[tokio::test]
+    async fn state() {
+        let server = Server::new(MockHandle {});
+        let fut = {
+            let mut io = IoHandler::new();
+            io.extend_with(server.to_delegate());
+            let (client, server) = local::connect::<Client, _, _>(io);
+            client.state().join(server)
+        };
+        let (have, _) = fut.compat().await.unwrap();
+        let want = serde_json::from_str(LIGHTBLOCK_JSON).unwrap();
+
+        assert_eq!(have, want);
+    }
+
+    struct MockHandle;
+
+    impl Handle for MockHandle {
+        fn latest_trusted(&self) -> Result<Option<LightBlock>, Error> {
+            let block: LightBlock = serde_json::from_str(LIGHTBLOCK_JSON).unwrap();
+
+            Ok(Some(block))
+        }
+    }
+
+    const LIGHTBLOCK_JSON: &str = r#"
```

Comment:
Is that needed twice? Here and above?

Reply:
The example is temporary to help with the integration work, wouldn't it to leave as is mid-term.